todo-mvc-kitchensink/src/widgets/App.ts

```diff
@@ -0,0 +1,194 @@
+import { deepAssign, assign } from '@dojo/core/lang';
+import uuid from '@dojo/core/uuid';
+import Map from '@dojo/shim/Map';
+import { v, w } from '@dojo/widget-core/d';
+import { I18nProperties, I18nMixin } from '@dojo/widget-core/mixins/I18n';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import pirateThemeStyles from '../themes/pirate';
+import { AppFooter } from './AppFooter';
+import Home from './Home';
+import * as styles from './styles/App.css';
+import { ThemeSwitcher } from './ThemeSwitcher';
+import TodoDetails from './TodoDetails';
+
+interface AppProperties extends ThemeableProperties, I18nProperties {
+	activeFilter?: string;
+	activeView?: string;
+	showDetails?: string;
+	pirateTheme?: boolean;
+}
+
+export interface Todo {
+	id: string;
+	label?: string;
+	completed?: boolean;
+	createdOn?: Date;
+}
+
+@theme(styles)
+export class App extends I18nMixin(ThemeableMixin(WidgetBase))<AppProperties> {
+	private _todos: Map<string, Todo> = new Map<string, Todo>();
+	private _todoItem: string = '';
+	private _completedCount: number = 0;
+	private _search: string = '';
+	private _updated: string = uuid();
+
+	constructor() {
+		super();
+
+		this.applyTheme();
+	}
+
+	applyTheme() {
+		this.setProperties(deepAssign({}, this.properties, {
```

Comment:
Even though this will kind of work, we shouldn't be calling `setProperties` from inside of a widget. We should probably update number 5 of https://github.com/dojo/widget-core#key-principles to reflect that.